conjur/wrapper/keystore_adapter.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+"""
+KeystoreAdapter module
+
+This module is a wrapper for the third-party keyring module used to get and set
+credentials in a system's keyring. This module allows us to easily swap out both
+library and functionality should we need to do so in the future.
+"""
+
+# Third party
+import keyring
+
+# Internals
+from conjur.constants import KEYSTORE_ATTRIBUTES
+
+
+class KeystoreAdapter:
+    """
+    KeystoreAdapter
+
+    A class for wrapping used functionality from the keyring library
+    """
+
+    @classmethod
+    def set_password(cls, identifier, key, val):
+        """
+        Method for setting a password in keyring
+        """
+        keyring.set_password(identifier, key, val)
+
+    @classmethod
+    def get_password(cls, identifier, key):
+        """
+        Method for getting a password in keyring
+        """
+        return keyring.get_password(identifier, key)
+
+    # pylint: disable=try-except-raise
+    @classmethod
+    def delete_password(cls, identifier):
```

Comment:
not sure this implementation is for the adapter. any case we should add simple delete for one attribute only here in the adapter

Reply:
for adapter because we don't want to raise an error on each password that doesnt exist. rather for the whole suite of credentials that dont exist

Reply:
i will make a note anyways for reference